x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/processors/user_agent.tsx

```diff
@@ -78,7 +95,39 @@ export const UserAgent: FunctionComponent = () => {
           'xpack.ingestPipelines.pipelineEditor.userAgentForm.propertiesFieldHelpText',
           { defaultMessage: 'Properties added to the target field.' }
         )}
-        propertyOptions={propertyOptions}
+        euiFieldProps={{
+          options: propertyOptions as [],
+          noSuggestions: !propertyOptions,
+          'data-test-subj': 'propertiesValueField',
+        }}
+      />
+
+      <UseField
+        config={fieldsConfig.extract_device_type}
+        component={ToggleField}
+        path="fields.extract_device_type"
+        data-test-subj="extractDeviceTypeSwitch"
+        euiFieldProps={{
+          label: (
+            <EuiFlexGroup gutterSize="xs">
+              <EuiFlexItem grow={false}>
+                <FormattedMessage
+                  id="xpack.ingestPipelines.pipelineEditor.userAgentForm.extractDeviceNameFieldText"
+                  defaultMessage="Extract Device Type"
+                />
+              </EuiFlexItem>
+              <EuiFlexItem style={{ marginBottom: -4 }}>
```

Comment:
This one feels a bit strange to me, but seemed like an acceptable trade off while we wait until the `size` prop to be available in the `EuiBetaBadge` component. If we don't apply the negative margin we end up with a ton of space between the `helpText` and the field it self, making it look different from the other fields:

With negative margin:
![Screenshot 2021-05-31 at 11 46 53](https://user-images.githubusercontent.com/1191206/120174927-f2427a00-c205-11eb-9b75-08cab0f216d6.png)


Without:
![image](https://user-images.githubusercontent.com/1191206/120174957-f9698800-c205-11eb-9948-16610f95a462.png)

Reply:
👍  thanks for the explanation and screenshots!

Can we check with the EUI team to see when the `size` prop will be available in Kibana? I see the [PR](https://github.com/elastic/eui/pull/4798) was merged a few days ago in EUI. If it's before 7.14, perhaps we can hold off on the hack.

I'd prefer not to have to add this inline style. But if we do need to proceed with this temporarily, let's open up an issue to make sure we don't forget to revisit.

Reply:
Tiny update: there's a [draft PR](https://github.com/elastic/kibana/pull/101334) for upgrading EUI in kibana. But greg's on PTO all week, so hopefully next week they will get it merged.

Reply:
Finally got merged, I've removed the inline styles and used the new size prop instead https://github.com/elastic/kibana/pull/100986/commits/68ac6135a6fb9a8ae1361324aa38226f65f27b34